tracer/src/Datadog.Trace/RemoteConfigurationManagement/Transport/RemoteConfigurationApi.cs

```
@@ -0,0 +1,66 @@
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(RemoteConfigurationApi));
+
+        private readonly IApiRequestFactory _apiRequestFactory;
+        private readonly IDiscoveryService _discoveryService;
+
+        private RemoteConfigurationApi(IApiRequestFactory apiRequestFactory, IDiscoveryService discoveryService)
+        {
+            _apiRequestFactory = apiRequestFactory;
+            _discoveryService = discoveryService;
+        }
+
+        public static RemoteConfigurationApi Create(IApiRequestFactory apiRequestFactory, IDiscoveryService discoveryService)
+        {
+            return new RemoteConfigurationApi(apiRequestFactory, discoveryService);
+        }
+
+        public async Task<GetRcmResponse> GetConfigs(GetRcmRequest request)
+        {
+            var uri = _apiRequestFactory.GetEndpoint(_discoveryService.ConfigurationEndpoint);
+            var apiRequest = _apiRequestFactory.Create(uri);
+
+            var requestContent = JsonConvert.SerializeObject(request);
+            var bytes = Encoding.UTF8.GetBytes(requestContent);
+            var payload = new ArraySegment<byte>(bytes);
+
+            var apiResponse = await apiRequest.PostAsync(payload, MimeTypes.Json).ConfigureAwait(false);
+            var isRcmDisabled = apiResponse.StatusCode == 404;
+            if (isRcmDisabled)
+            {
+                Log.Debug("Remote Configuration has been disabled.");
+                return null;
+            }
+
+            var content = await apiResponse.ReadAsStringAsync().ConfigureAwait(false);
+            if (apiResponse.StatusCode is not (>= 200 and <= 299))
+            {
+                Log.Warning<int, string>("Failed to receive remote configurations {StatusCode} and message: {ResponseContent}", apiResponse.StatusCode, content);
+
+                return null;
+            }
+
+            var response = JsonConvert.DeserializeObject<GetRcmResponse>(content);
```

might want a try catch around this?

---

Hmmm, I'm not sure.
Due to RCM spec, we should report errors, so we want to report this one, if deserialization is failed.
If we add try..catch, the exception will be swallowed inside of `Api`

WDYT?

---

Yeah, we want the exception to bubble up here, so leaving it to throw is fine

---

I have `try..catch` block at the top level of `PollAsync` method, so we don't need something specific here.